src/App.tsx

```diff
@@ -1,61 +1,160 @@
+  const largestId = usedIds[usedIds.length - 1];
+  const newId = largestId + 1;
+
+  return newId;
+};
+
+const initialTodoList: PreparedTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(usersFromServer, todo.userId),
+}));
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userIdSelect, setUserIdSelect] = useState(0);
+  const [preparedTodoList, setPreparedTodoList] = useState(initialTodoList);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleEnter = (value: string) => {
+    setTitleInput(value);
+    if (titleError) {
+      setTitleError(false);
+    }
+  };
+
+  const handleUserSelect = (value: number) => {
+    setUserIdSelect(value);
+    if (userError === true) {
+      setUserError(false);
+    }
+  };
+
+  const validateForm = () => {
+    const isNoTitle = titleInput === '';
+    const isNoUser = userIdSelect === 0;
+
+    if (isNoTitle || isNoUser) {
+      if (isNoTitle) {
+        setTitleError(true);
+      }
+
+      if (isNoUser) {
+        setUserError(true);
+      }
+
+      return false;
+    }
+
+    return true;
+  };
+
+  const clearForm = () => {
+    setTitleInput('');
+    setUserIdSelect(0);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!validateForm()) {
+      return;
+    }
+
+    const id = getNewTodoId(preparedTodoList);
+    const user = findUser(usersFromServer, userIdSelect);
+
+    setPreparedTodoList(currentList => (
+      [
+        ...currentList,
+        {
+          id,
+          title: titleInput,
+          completed: false,
+          user,
+        },
+      ]
+    ));
+
+    clearForm();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title: '}
+
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={titleInput}
+              onChange={event => handleTitleEnter(event.target.value)}
+            />
+          </label>
+
+          {titleError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            {'User: '}
+
+            <select
+              data-cy="userSelect"
+              value={userIdSelect}
+              onChange={event => handleUserSelect(Number(event.target.value))}
```

Comment:
Just take in `handleUserSelect` a whole event & make some logic in this fucntion
```suggestion
              onChange={handleUserSelect}
```